src/test/java/com/amazon/ion/impl/bin/IonManagedWriter_1_1_Test.kt

```diff
@@ -733,11 +739,92 @@ internal class IonManagedWriter_1_1_Test {
         assertEquals(expected, actual)
     }
 
+    @Test
+    fun `writeObject() should write something with nested macro representation`() {
+        val expected = """
+            $ion_1_1
+            $ion_encoding::((macro_table (macro null (x*) x) (macro Polygon (vertices+ compact_symbol::fill?) {vertices:[vertices],fill:(0 fill)}) (macro Point2D (x y) {x:x,y:y})))
+            (:Polygon (: (:Point2D 0 0) (:Point2D 0 1) (:Point2D 1 1) (:Point2D 1 0)) Blue)
+        """.trimIndent()
+
+        val data = Polygon(
+            listOf(
+                Point2D(0, 0),
+                Point2D(0, 1),
+                Point2D(1, 1),
+                Point2D(1, 0),
+            ),
+            Colors.Blue,
+        )
+
+        val actual = write {
+            writeObject(data)
+        }
+
+        assertEquals(expected, actual)
+    }
+
+    private data class Polygon(val vertices: List<Point2D>, val fill: Colors?) : WriteAsIon {
+        init { require(vertices.size >= 3) { "A polygon must have at least 3 edges and 3 vertices" } }
+
+        companion object {
+            // This is a very long macro name, but by using the qualified class name,
+            // there is almost no risk of having a name conflict with another macro.
+            private val MACRO_NAME = Polygon::class.simpleName!!.replace(".", "_")
```

Comment:
When using the simple name, this comment is no longer accurate, right?

Reply:
You are correct.
```suggestion
            // Using the qualified class name would be verbose, but may be safer for general
            // use so that there is almost no risk of having a name conflict with another macro.
            private val MACRO_NAME = Polygon::class.simpleName!!.replace(".", "_")
```